src/durableorchestrationcontext.ts

```diff
@@ -293,9 +334,25 @@ export class DurableOrchestrationContext {
      * @returns A TimerTask that completes when the durable timer expires.
      */
     public createTimer(fireAt: Date): TimerTask {
-        const newAction = new CreateTimerAction(fireAt);
-        const task = new DFTimerTask(false, newAction);
-        return task;
+        const timerAction = new CreateTimerAction(fireAt);
+        const durationUntilFire = moment.duration(moment(fireAt).diff(this.currentUtcDateTime));
+        if (
+            this.schemaVersion >= ReplaySchema.V3 &&
+            this.maximumShortTimerDuration &&
+            this.longRunningTimerIntervalDuration &&
+            durationUntilFire > this.maximumShortTimerDuration
+        ) {
+            return new LongTimerTask(
+                false,
+                timerAction,
+                this,
+                this.taskOrchestratorExecutor,
+                this.maximumShortTimerDuration,
+                this.longRunningTimerIntervalDuration
+            );
+        }

```

Comment:
I think it'd be wise to protect against the case where ReplaySchema is V3 but the maximumShortTimerDuration and/or the longRunningTimerIntervalDuration are undefined.

Can we add a check here that throws an exception if that's the case? Otherwise, we risk things going wrong in the extension-level.

Reply:
I was actually going to ask for feedback regarding this `if` statement. Should we even be checking for schema version? If the schema is less than `V3`, an exception would be thrown in the extension, regardless of whether we create a `LongTimerTask` or a `DFTimerTask`. 

Also, I was going to ask what should we do in the case that `maximumShortTimerDuration` is defined but not `longRunningTimerIntervalDuration`? Should we also complain/throw an exception? Just ignore and create a `DFTimerTask`? Default to using `maximumShortTimerDuration` as the `longRunningTimerIntervalDuration`?

Reply:
Add the check! On second thought, I actually think we need to wrap this in a check for the replay version if we want to throw this error. We don't want to complain about timer-related metadata not being set if schema version < 3 is being used anyway.

The alternative I guess would be to ignore this case entirely and let it error out in the extension, but I think throwing the error here is more helpful.

Also, would appreciate thoughts on the error message 😛

Reply:
Left some comments on the error message in my latest review!